backend/src/components/reports.js

```diff
@@ -41,6 +52,139 @@ async function getFacilityReports(req, res) {
   }
 }
 
+async function getSurveySections(req, res) {
+  try {
+    const sections = []
+    const operation = `ofm_sections?$filter=ofm_is_published eq true and _ofm_survey_value eq '${req?.query?.surveyId}'&$orderby=ofm_section_order`
+    const response = await getOperation(operation)
+    response?.value?.forEach((section) => sections.push(new MappableObjectForFront(section, SurveySectionMappings).toJSON()))
+    return res.status(HttpStatus.OK).json(sections)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function getSurveyQuestions(req, res) {
+  try {
+    const questions = []
+    let operation
+    if (req?.query?.sectionId) {
```

Comment:
I agree that it will throw error if the sectionId is missing in the query from frontend. However, in the frontend, I've added a check to only call this api if there is a sectionId. Also, I think it would be better to throw error than returning all questions if a sectionId is missing. Please let me know what you think.

Reply:
I'm just a little confused because the comment on the route says "Get survey's questions using query (sectionId for section's questions, questionId for question's column headers)". I would think either we need to accept both params or add validation in the route for sectionId.

Reply:
sorry it is my bad. I did some code refactoring to simplify logic and I forgot to update my old comments. It is updated now. Also, the reason I used query instead of params is because I wanted to have more flexibility to update this api in the future (if needed)

Reply:
I can also throw an error if the query is empty, sth like below:
throw new ApiError(HttpStatus.BAD_REQUEST, { message: 'API query is missing' })

Reply:
If we're expecting at least one query param (assuming more are possible) and don't want to execute with no params (since this would return too much data) then we definitely want some validation, either in the route or component. Whatever works best.